### PR TITLE
fix(sso): handle multiple regions

### PR DESCRIPTION
**Pull request by frenautvh**

### CODE REVIEW

File: webpack.dev.js [hunk 1]
```diff
@@ -1,20 +1,27 @@
...
-  logLevel: 'silent',
+const TARGET = {
+  eu: 'https://www.ovh.com',
+  ca: 'https://ca.ovh.com',
+  us: 'https://us.ovhcloud.com',
 };
 
 module.exports = (env) => {
+  const region = (env.region || 'eu').toLowerCase();
+  const proxyOptions = {
```
Comment (cbourgois) [inline webpack.dev.js]: We have created `/server/proxy/aapi` to isolate config, maybe we should move this part in `/server/proxy/***` .

Comment (frenautvh) [reply webpack.dev.js]: yep i'm going to move it

File: server/sso/index.js [hunk 2]
```diff
@@ -3,124 +3,137 @@ const base64url = require('base64url');
...
+          console.error('[SSO] - crosslogin - error: ', err);
+          return resp.status(500);
         }
-        if (parsedCookie.USERID) {
-          res.cookie('USERID', parsedCookie.USERID, { path: '/' });
+
+        const cookies = resp.headers['set-cookie'];
+        let parsedCookie;
+
+        for (let i = cookies.length - 1; i >= 0; i -= 1) {
```
Comment (FredericEspiau) [inline server/sso/index.js]: Can you maybe `cookies.reverse().map(cookie => ...)` ?

File: server/sso/index.js [hunk 3]
```diff
@@ -3,124 +3,137 @@ const base64url = require('base64url');
...
+      baseUrl: 'https://www.ovh.com/cgi-bin/crosslogin.cgi',
+      devLoginUrl: 'https://www.ovh.com/auth/requestDevLogin/',
+    },
+    ca: {
+      host: 'ca.ovh.com',
+      devLoginUrl: 'https://ca.ovh.com/auth/requestDevLogin/',
+      baseUrl: 'https://ca.ovh.com/cgi-bin/crosslogin.cgi',
+    },
+    us: {
+      host: 'ovhcloud.com',
```
Comment (FredericEspiau) [inline server/sso/index.js]: No `us.` ? Could be nice to have some comments to understand why then 😄

File: server/sso/index.js [hunk 4]
```diff
@@ -3,124 +3,137 @@ const base64url = require('base64url');
...
 
-const config = {
+const CONFIG = {
   ssoAuth: {
-    host: 'www.ovh.com',
-    baseUrl: 'https://www.ovh.com/cgi-bin/crosslogin.cgi',
-    devLoginUrl: 'https://www.ovh.com/auth/requestDevLogin/',
+    eu: {
+      host: 'www.ovh.com',
+      baseUrl: 'https://www.ovh.com/cgi-bin/crosslogin.cgi',
```
Comment (FredericEspiau) [inline server/sso/index.js]: It can be guessed from the `host`, can't it ?

File: server/sso/index.js [hunk 5]
```diff
@@ -3,124 +3,137 @@ const base64url = require('base64url');
...
+      devLoginUrl: 'https://www.ovh.com/auth/requestDevLogin/',
```
Comment (FredericEspiau) [inline server/sso/index.js]: It can be guessed from the `host`, can't it ?

File: server/sso/index.js [hunk 6]
```diff
@@ -3,124 +3,137 @@ const base64url = require('base64url');
...
 
-  try {
-    cookies = JSON.parse(base64url.decode(req.query.data));
+    try {
+      cookies = JSON.parse(base64url.decode(req.query.data));
 
-    if (Array.isArray(cookies.cookies)) {
-      cookies.cookies.forEach((c) => {
-        const parsedCookie = cookie.parse(c);
+      if (Array.isArray(cookies.cookies)) {
```
Comment (FredericEspiau) [inline server/sso/index.js]: The cookies have cookies themselves ? I guess this is something we don't have control on ?

File: server/sso/index.js (hunk 3 above)
Comment (frenautvh) [reply server/sso/index.js]: good catch

File: server/sso/index.js (hunk 2 above)
Comment (frenautvh) [reply server/sso/index.js]: cookies.reverse().forEach(...) :p

**Pull request merged**